test/test_token_count.py

```diff
@@ -0,0 +1,67 @@
+from autogen.token_count_utils import count_token, num_tokens_from_functions
+
+
+def test_function_token_count():
+    func1 = {
+        "name": "sh",
+        "description": "run a shell script and return the execution result.",
+        "parameters": {
```

Comment:
if the token counting logic changes or if the model that interprets the tokens changes. A change in the underlying logic or model would require updating all the hard-coded numbers, which is error-prone and not scalable.

```python
import pytest
@pytest.mark.parametrize("input_functions, expected_count", [
    ([func1], 44),
    ([func2], 47),
    ([func3], 45),
    ([func1, func2], 79)
])

def test_function_token_count(input_functions, expected_count):
    assert num_tokens_from_functions(input_functions) == expected_count
```

Also, by conventions you should provide descriptive test function names, using the name of the function your testing:

- `test_num_tokens_from_functions`, not `test_function_token_count`

Reply:
Thanks,  I am changing this.